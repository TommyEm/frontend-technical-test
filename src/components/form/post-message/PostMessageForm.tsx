import { FC, useCallback, useState } from 'react'

import { Input } from '../input/Input'
import { IconButton } from '@/components/button/icon/IconButton'
import { usePostMessage } from '@/hooks/useMessages'
import { loggedUserId } from '@/pages/_app'
import styles from './PostMessageForm.module.css'


export interface IPostMessageFormProps {
	conversationId: number
}

export const PostMessageForm: FC<IPostMessageFormProps> = ({
	conversationId,
}) => {
	const [newMessage, setNewMessage] = useState<string>('')
	const { mutate: postMessage } = usePostMessage(conversationId)

	const handleChangeNewMessage = useCallback((e) => {
		setNewMessage(e.currentTarget.value)
	}, [])

	const handleSubmit = (event) => {
		event.preventDefault()

		if (newMessage !== '') {
			postMessage({
				authorId: loggedUserId,
				message: newMessage,
			})
			setNewMessage('')
		}
	}

	return (
		<form
			className={styles.container}
			onSubmit={handleSubmit}
		>
			<div className={styles.input}>
				<Input
					value={newMessage}
					onChange={handleChangeNewMessage}
				/>
			</div>
			<div className={styles.button}>
				<IconButton
					label='Send'
					onClick={handleSubmit}
				/>
			</div>
		</form>
	)
}
