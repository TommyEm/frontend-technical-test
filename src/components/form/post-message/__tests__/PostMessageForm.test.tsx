import {
	cleanup,
	render,
	screen,
} from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import { composeStories } from '@storybook/testing-react'

import * as stories from '../PostMessageForm.stories'

const { Default } = composeStories(stories)
const queryClient = new QueryClient()

afterEach(cleanup)
const emptyComponent = (
    <QueryClientProvider client={queryClient}>
		<Default />
    </QueryClientProvider>
)

it('should render a "post message" form', () => {
	render(emptyComponent)
	expect(screen.getByRole('textbox')).toBeVisible()
	expect(screen.getByRole('button')).toBeVisible()
})

it('should match snapshot', () => {
	const tree = render(emptyComponent)
	expect(tree).toMatchSnapshot()
})
