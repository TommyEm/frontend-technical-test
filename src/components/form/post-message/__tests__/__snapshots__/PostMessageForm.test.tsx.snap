// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="container"
      >
        <div
          class="input"
        >
          <input
            class="input"
            data-testid="test-Input"
            placeholder=""
            type="text"
            value=""
          />
        </div>
        <div
          class="button"
        >
          <button
            class="button"
          >
            <div
              class="label"
            >
              Send
            </div>
            <div
              class="icon"
            >
              <span
                class="container mod-primary"
                data-testid="test-Icon"
              >
                <svg
                  class="icon"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16,464,496,256,16,48V208l320,48L16,304Z"
                  />
                </svg>
              </span>
            </div>
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="container"
    >
      <div
        class="input"
      >
        <input
          class="input"
          data-testid="test-Input"
          placeholder=""
          type="text"
          value=""
        />
      </div>
      <div
        class="button"
      >
        <button
          class="button"
        >
          <div
            class="label"
          >
            Send
          </div>
          <div
            class="icon"
          >
            <span
              class="container mod-primary"
              data-testid="test-Icon"
            >
              <svg
                class="icon"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16,464,496,256,16,48V208l320,48L16,304Z"
                />
              </svg>
            </span>
          </div>
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
